Update Tencent documentation (#4781)
mendix/docs: content/en/docs/developerportal/deploy/tencent-deploy/_index.md

* A Siemens Low Code Platform app created with Studio or Studio Pro

## 3 Cluster and Namespace Management{#cluster-namespace}

### 3.1 Cluster Overview {#overview}

Go to the Cluster Manager page by clicking the **切换到** menu and selecting **集群管理**.

If you have not got any clusters, you will first have to purchase one. Alternatively, you can use a free trial cluster to try out the Siemens Low-Code Platform experience, or register your own cluster (running Kubernetes versions 1.19 to 1.22).

* Click **[前往腾讯云门户网站](https://cloud.tencent.com/solution/slp)** to go to Tencent and purchase the necessary Siemens Low-Code Platform on Tencent Cloud resources.
* Click **请求访问免费的应用集群** to request access to a free cluster to try out Siemens Low-Code Platform. You will need to create an app first, see [App & Team Management](/not/a/good/url) for more information.
* Click **注册集群** to register an existing cluster to use for deploying your app.

When you have one or more clusters, you can see a summary of them with all their namespaces and an indication of the namespace status and how long it has been running (**上线时间**).

> {{< figure src="/attachments/developerportal/deploy/tencent-deploy/cluster-overview.png" >}}

Reviewer: Not sure if we have discussed this – shall we also replace this screenshot with its Chinese version?

Author: In my build, this is the Chinese version?

Each cluster resource you purchase from Tencent automatically contains a number of namespaces and you will run your Siemens Low Code Platform app inside one of these namespaces. The namespaces are designed to hold separate test, acceptance, and production versions of your app so that resource sharing between the versions is minimized.

You can also see the activities logged for all you clusters by clicking **活动** at the top of the page. This shows the following:

* When a cluster has been added
* When a cluster description is added
* When name of the cluster is changed
* When cluster description is changed

The cluster overview will refresh automatically, but you can force a refresh by clicking **刷新** at the top of the page.

### 3.2 Cluster Details

Click **详情** next to the cluster name to see information about the entire cluster.

In the **命名空间** tab you can see the namespaces in the cluster and their current status. This duplicates the information you can see in the cluster overview.

> {{< figure src="/attachments/developerportal/deploy/tencent-deploy/cluster-details.png" >}}

Reviewer: Same as above.

Author: In my build, this is the Chinese version?

There are two other tabs:

* **集群管理员** – where you can see other people who have access to the cluster and can manage them
* **活动** – where you can see all the activities which have taken place on this cluster

#### 3.2.1 Members

The **集群管理员** tab will show you all the cluster managers for this cluster. **状态** indicates whether the user has accepted an invitation.

> 

Reviewer: Maybe also add info about **移除**, which means to remove a cluster manager from the cluster.

## 1 Introduction

> To deploy apps to Tencent, you will first have to purchase your cloud resources. You can then manage these resources from within the Mendix ATM as described below in [Cluster and Namespace Management](#cluster-namespace).

Reviewer: Maybe "App & Team Management (ATM)" instead?

Author: Good idea - and I should really change all references to **Mendix** to be **Siemens Low Code Platform**

* Mendix Studio Pro China Edition

    * You will be able to download Mendix Studio Pro, as the first step of the onboarding during your initial sign up:
    * You can also get a copy from the **下载 Studio Pro** option in the **切换到** menu at the top left of **App & Team Management**.

* A Mendix app created with Studio Pro

> ## 3 Cluster and Namespace Management{#cluster-namespace}

Author: Need more access to finish documenting this section

* Mendix Studio Pro China Edition

>     * You will be able to download Mendix Studio Pro, as the first step of the onboarding during your initial sign up:

Reviewer: or should it be "...during your initial sign up."  ?

    * You can also get a copy from the **下载 Studio Pro** option in the **切换到** menu at the top left of **App & Team Management**.

* A Mendix app created with Studio Pro

## 3 Cluster and Namespace Management{#cluster-namespace}

### 3.1 Cluster Overview {#overview}

Go to the Cluster Manager page by clicking the **切换到** menu and selecting **集群管理**.

If you have not got any clusters, you will first have to purchase one. Alternatively, you can use a free trial cluster to try out the Siemens Low-Code Platform experience, or register your own cluster (running Kubernetes versions 1.19 to 1.22).

* Click **[前往腾讯云门户网站](https://cloud.tencent.com/solution/slp)** to go to Tencent and purchase the necessary Siemens Low-Code Platform on Tencent Cloud resources.
* Click **请求访问免费的应用集群** to request access to a free cluster to try out Siemens Low-Code Platform. You will need to create an app first, see [App & Team Management](/not/a/good/url) for more information.
* Click **注册集群** to register an existing cluster to use for deploying your app.

{{% todo %}}Update UX descriptions and confirm flows{{% /todo %}}

> When you have one or more clusters, you can see a summary of them with all their namespaces and an indication of the namespace status and how long it has been running (Uptime).

Reviewer: Should "Uptime** be in bold and in Chinese?

{{< figure src="/attachments/developerportal/deploy/tencent-deploy/cluster-overview.png" >}}

> Each cluster resource you purchase from Tencent automatically contains a number of *命名空间* and you will run your Mendix app inside one of these namespaces. The namespaces are designed to hold separate test, acceptance, and production versions of your app so that resource sharing between the versions is minimized.

Reviewer: Should 命名空间 be in bold?

Author: Not sure - in this case it isn't a UX element. So, perhaps, it shouldn't even be in Chinese?

Author: Keep in English and not italicize

You can also see the activities logged for all you clusters by clicking **活动** at the top of the page. This shows the following:

* When a cluster has been added
* When a cluster description is added
* When name of the cluster is changed
* When cluster description is changed

The cluster overview will refresh automatically, but you can force a refresh by clicking **刷新** at the top of the page.

### 3.2 Cluster Details

Click **详情** next to the cluster name to see information about the entire cluster.

> In the **命名空间** tab you can see the **Namespaces** in the cluster and their current status. This duplicates the information you can see in the cluster overview.

Reviewer: If **Namespace** here is UI text, should it be in Chinese?

Author: I'll just use namespaces and not use the UI text (查看命名空间 - which Chrome translates as "View Namespaces", not just "Namespaces").

{{< figure src="/attachments/developerportal/deploy/tencent-deploy/cluster-details.png" >}}

There are two other tabs:

> * 集群管理员 – where you can see other people who have access to the cluster and can manage them

Reviewer: Should tab names be bold?

* cluster managers who are added to the cluster cannot remove the cluster manager who purchased the cluster resources
{{% /alert %}}

### 3.3 Namespace Management

If you are a member of a namespace, you can also manage a namespace in the cluster.

> Click the **详情** button for the namespace you want to manage.

Reviewer: Maybe "Click **详情** for the namespace you want to manage."

On the namespace management page, there are a number of tabs which allow you to manage aspects of your namespace :

> * 应用 – all the apps deployed in this namespace

Reviewer: Should all the tab names be bold?

* When a user is removed as a namespace member
* When user's permission is changed in the namespace

#### 3.3.1 Apps

The **应用** tab of namespace details lists all the apps which are deployed to this namespace.

> {{< figure src="/attachments/developerportal/deploy/tencent-deploy/namespace-apps.png" >}}

Reviewer: Same as above.

Author: In my build, this is the Chinese version?

> Click **配置** to make Pod, Ingress, or service annotations or configure runtime metrics for this environment.

Reviewer: Do we want to explain the page that opens after you click **配置** here?

Author: Good idea,
I guess it will be similar to https://docs.mendix.com/developerportal/deploy/private-cloud-cluster/#6211-configure-environment and https://docs.mendix.com/developerportal/deploy/private-cloud-cluster/#operate, but I'm not sure what it will look like.
I'll pick this up early next week.

Reviewer: Indeed, it is very similar to https://docs.mendix.com/developerportal/deploy/private-cloud-cluster/#6211-configure-environment.

* When a user is removed as a namespace member
* When user's permission is changed in the namespace

#### 3.3.1 Apps

> The **成员** tab of namespace details lists all the apps which are deployed to this namespace.

Reviewer: **成员** means "members", so the translation here is not correct.

Author: How would I manage without you to spot these things!
How about 应用?

* Manage Custom Runtime Settings
* Manage Log levels

The **成员** tab allows you to manage the list of members of the namespace and control what rights they have.

##### 3.3.2.1 Adding Members

You can invite additional members to the namespace, and configure their role depending on what they should be allowed to do.

> 1. The **成员** tab displays a list of current members (if any).

Reviewer: This is not a step, so perhaps it's better not to make it a listed item.

2. Click **邀请成员**.

3. Enter the **腾讯ID** of the person you want to invite.

4. Give them the rights they need. This can be:
   
    1. **开发人员** – a standard set of rights needed by a developer, these are listed on the screen
    2. **管理员** – a standard set of rights needed by an administrator, these are listed on the screen
    3. **自定义** – you can select a custom set of rights by checking the box next to each role you want to give to this person

5. Click **发送邀请** to send an invite to this person.

6. The next time the user signs in to the Mendix ATM, they will be added to the namespace.

##### 3.3.2.2 Editing & Removing Members

You can remove existing members.

{{% todo %}}What is remove member?{{% /todo %}}

1. On the **成员** tab of the namespace details, click **Remove** next to the member you want to remove.

If you want to change the rights for an existing member, you will have to remove them and then add them again with the updated rights.

#### 3.3.3 Operate {#operate}

The **操作** tab allows you to add a set of links which are used when users request a page from the Operate category for their app in the ATM, as shown below.

The following pages can be configured:

> * 指标

Reviewer: Not sure if these page names are UI text. If they are, maybe they should be bold?

{{% /alert %}}

{{% alert color="warning" %}}
There is a limit of 200MB on the size of a deployment package which can be deployed on the Tencent Cloud.
{{% /alert %}}

1. On the **环境概览** page for your app in the ATM, click **从 Team Server 创建包**.
   
2. Select the branch which contains the commit for which you want to create a deployment package and click **下一步**.
   
    {{< figure src="/attachments/developerportal/deploy/tencent-deploy/image10.png" >}}

3. Select the revision/commit for which you want to create a deployment package and click **下一步**.
   
    {{< figure src="/attachments/developerportal/deploy/tencent-deploy/image11.png" >}}

4. Enter a **新版本** and **标签描述** according to your own deployment procedure.

5. If you want to deploy and start your package immediately, check **自动部署** and select an existing environment in **用于自动部署的环境**. You need to make sure that the environment is ready using the techniques described in [Deploying the Deployment Package](#deploy-package), below, where you can also see how to deploy a deployment package manually.

> 6. Click **构建此修订版**

Reviewer: A period can be added.

If you request access to a free cluster you will not need to create your own environment, this will be done for you as part of the request you made. An environment will be created for you using the latest MDA package you created in [Creating a Deployment Package](#create-deployment-package), above.

You can deploy updated deployment packages by following the instructions in [Deploying the Deployment Package](#deploy-package), below.
{{% /alert %}}

When deploying your app for the first time, there will be no environment to deploy to. To create an environment, you will need to create a deployment package as described in [Creating a Deployment Package](#create-deployment-package), above.

1. Go to the **环境概览** page for your app in the ATM.

2. Click **创建环境**.

    If there are no clusters available to you, you will have to create one or get access to an existing one before you can continue. You will be taken to the Cluster Manager page. See [Cluster and Namespace Management](#cluster-namespace), above, for information on the three ways you can get a cluster, then start this section again.

2. Choose the deployment package you want to use to create the environment and click **下一步**.

> 2. A **内部名称** will be generated for you. This will be used when creating your environment to ensure that all the environment names in your namespace are unique.

Reviewer: This is not a step, so perhaps it's better not to make it a listed item.

    2. For production deployment, select **生产环境**. If you select **生产环境**, then you will be asked for the **订阅密钥** which you received when you purchased Tencent resources.

        {{% alert color="warning" %}}Your app can only be deployed to a production environment if security is set on. You will not receive an error if security is set off, but the deployment will appear to hang with a spinner being displayed.{{% /alert %}}

6. Click **下一步**.

7. Select **核心资源**.

    For core resources, there are two sets of values. The **请求** value is the amount of core resources which are initially requested. The **限制** value is the maximum amount of resource that the environment can use.

    There are three pre-defined sets of resources, **小**, **中等**, and **大**. Choosing these will set the **CPU** and **Memory** values automatically.

    | **Name** | **CPU cores**: Limit | **Memory (Gb)**: Limit | **CPU cores**: Request | **Memory (Gb)**: Request |
    | --- | --- | --- | --- | --- |
    | Small | 1 | 0.5 | 0.1 | 0.5 |
    | Medium | 2 | 2 | 1 | 1 |
    | Large | 4 | 4 | 2 | 2 |
    | Custom | own choice | own choice | own choice | own choice |

>     Alternatively, you can choose **自定义**, and enter your own requirements for **CPU** and **内存**. Ensure that these values are the same or greater than the values for a *小* environment, otherwise you may run into problems running your app.

Reviewer: Is the italics style of *小* here for emphasis? (If it is UI text, it should be bold.)

Author: It's emphasis, but not needed, I don't think. I'll remove it.

8. Select a **数据库计划** from the list of plans in the dropdown. This will have the format `db-plan-…` where the suffix reflects the environment.

9. Select a **文件存储计划** from the list of plans in the dropdown. This will have the format `file-plan-…` where the suffix reflects the environment.

10. Click **创建环境**.

> 11. You will see your new environment listed. An *in-progress* icon will be shows next to the resource plans until they have been provisioned.

Reviewer: This listed item is not an action/step. (Maybe use "Check your new environment in the list"?)

1. Click **部署** next to the deployment package you wish to deploy.

2. Confirm the **目标** environment (you can select a different one here if one is available).

3. Confirm that the **状态** is *准备好*.

4. Click **传输**.
   
    {{< figure src="/attachments/developerportal/deploy/tencent-deploy/image15.png" >}}

5. Change any constants in the **常量** tab: select the constant you want to edit and then click **编辑**.

6. Toggle any scheduled events in the **计划事件** tab: select the scheduled event you want to enable or disable and click **启停**.

7. Click **继续** to continue to the Start Application confirmation page.

8. Click **应用更改** to deploy the application to the selected environment. The app will start automatically once the deployment is successful.

You will be taken to the Environment Details page for the selected environment. You can find information about managing your environment from this page in [Managing Your Environments from the Environment Details Page](#environment-details), below.

## 5 Environments Page

The **环境概览** page of the ATM contains three sections:

> * Deployment Package Repository

Reviewer: These sections are named in Chinese in ATM as follows:
**部署包仓库
环境
活动日志**

Author: @Luyao-Zhang-1 So you think we should put them here in Chinese? Makes sense.
Perhaps we can discuss this on Monday?

5. Change any constants in the **常量** tab: select the constant you want to edit and then click **编辑**.

6. Toggle any scheduled events in the **计划事件** tab: select the scheduled event you want to enable or disable and click **启停**.

7. Click **继续** to continue to the Start Application confirmation page.

8. Click **应用更改** to deploy the application to the selected environment. The app will start automatically once the deployment is successful.

You will be taken to the Environment Details page for the selected environment. You can find information about managing your environment from this page in [Managing Your Environments from the Environment Details Page](#environment-details), below.

## 5 Environments Page

The **环境概览** page of the ATM contains three sections:

* **部署包仓库**
* **环境**
* **活动日志**

### 5.1 Deployment Package Repository

The **部署包仓库** section lists the deployment packages which have been created for this app.

{{< figure src="/attachments/developerportal/deploy/tencent-deploy/image17.png" >}}

There are five buttons:

* **刷新**
* **从 Team Server 创建包**
* **上传**
* **详情**
* **部署**

These are described in more detail below.

> #### 5.1.1 Refresh

Author: Need to insert a Delete section here and renumber.

Reviewer: I am not sure if we should replace the Refresh 

> Need to insert a Delete section here and renumber.

Please see my comment above. We might need to replace the **Refresh** section with the **Delete** section, depending on whether the **Refresh** button really exists or not.

* Manage Custom Runtime Settings
* Manage Log levels

##### 3.3.2.1 Adding Members

You can invite additional members to the namespace, and configure their role depending on what they should be allowed to do.

The **成员** tab displays a list of current members (if any).

1. Click **邀请成员**.

2. Enter the **腾讯ID** of the person you want to invite.

3. Give them the rights they need. This can be:
   
    1. **开发人员** – a standard set of rights needed by a developer, these are listed on the screen
    2. **管理员** – a standard set of rights needed by an administrator, these are listed on the screen
    3. **自定义** – you can select a custom set of rights by checking the box next to each role you want to give to this person

4. Click **发送邀请** to send an invite to this person.

5. The next time the user signs in to ATM, they will be added to the namespace.

##### 3.3.2.2 Editing & Removing Members

You can remove existing members.

{{% todo %}}What is remove member?{{% /todo %}}

> 1. On the **成员** tab of the namespace details, click **Remove** next to the member you want to remove.

Reviewer: The button on the Chinese version actually says **编辑** (Edit) here.  I tried the procedure. It is something like this:

1. On the **成员** tab of the namespace details, click **编辑** next to the member you want to edit or remove.
2. In the **编辑成员** dialog box, search the member by entering their email address.
3. To remove the member, click **移除成员**。
4. To edit the member, select or deselect the check boxes to change their privileges.
5. Click **保存** to save the changes.

Author: Thanks for finding that.
Can you add the process?
Stylistically I find it better to start each step with the action. e.g.

1. Search for the member in the **编辑成员** dialog box by entering their email address.
2. Click **移除成员** to remove the member.

 - but you don't have to if you feel it is better the other way around:

#### 3.3.5 Install

The **安装** tab allows you to download the Configuration Tool by specifying the operating system of the local computer on which you are going to run it.

#### 3.3.6 Additional Information

The **附加信息** tab displays information about the namespace, such as software versions (for example, the Operator version).

## 4 Deploying an App for the First Time

### 4.1 Creating a Deployment Package {#create-deployment-package}

Before you can deploy your app, you will need to create a deployment package. Ensure that you have committed the version of the app you want to deploy before continuing.

{{% alert color="info" %}}
Follow the instructions below to create a deployment package from the Team Server. Alternatively, if you already have an MDA package, you can upload it by clicking **上传**.
{{% /alert %}}

{{% alert color="warning" %}}
There is a limit of 200MB on the size of a deployment package which can be deployed on the Tencent Cloud.
{{% /alert %}}

1. On the **环境概览** page for your app in the ATM, click **从 Team Server 创建包**.
   
2. Select the branch which contains the commit for which you want to create a deployment package and click **下一步**.
   

>     {{< figure src="/attachments/developerportal/deploy/tencent-deploy/image10.png" >}}

Reviewer: Change it to the screenshot of the Chinese version?

3. Select the revision/commit for which you want to create a deployment package and click **下一步**.
   

>     {{< figure src="/attachments/developerportal/deploy/tencent-deploy/image11.png" >}}

Reviewer: Same as above.

4. Enter a **新版本** and **标签描述** according to your own deployment procedure.

5. If you want to deploy and start your package immediately, check **自动部署** and select an existing environment in **用于自动部署的环境**. You need to make sure that the environment is ready using the techniques described in [Deploying the Deployment Package](#deploy-package), below, where you can also see how to deploy a deployment package manually.

6. Click **构建此修订版**.
   

>     {{< figure src="/attachments/developerportal/deploy/tencent-deploy/image12.png" >}}

Reviewer: Same as above.

1. Click **下一步**.

1. Select **核心资源**.

    For core resources, there are two sets of values. The **请求** value is the amount of core resources which are initially requested. The **限制** value is the maximum amount of resource that the environment can use.

    There are three pre-defined sets of resources, **小**, **中等**, and **大**. Choosing these will set the **CPU** and **Memory** values automatically.

    | **Name** | **CPU cores**: Limit | **Memory (Gb)**: Limit | **CPU cores**: Request | **Memory (Gb)**: Request |
    | --- | --- | --- | --- | --- |
    | Small | 1 | 0.5 | 0.1 | 0.5 |
    | Medium | 2 | 2 | 1 | 1 |
    | Large | 4 | 4 | 2 | 2 |
    | Custom | own choice | own choice | own choice | own choice |

    Alternatively, you can choose **自定义**, and enter your own requirements for **CPU** and **内存**. Ensure that these values are the same or greater than the values for a 小 environment, otherwise you may run into problems running your app.

1. Select a **数据库计划** from the list of plans in the dropdown. This will have the format `db-plan-…` where the suffix reflects the environment.

> 1. Select a **文件存储计划** from the list of plans in the dropdown. This will have the format `file-plan-…` where the suffix reflects the environment.

Reviewer: On my screen, it is **存储计划** (without the first two characters).

Author: OK - probably an error in my copy/paste. Does it make sense without the first two characters 😀
```suggestion
1. Select a **存储计划** from the list of plans in the dropdown. This will have the format `file-plan-…` where the suffix reflects the environment.
```

Reviewer: Yes, it still makes sense. It means "storage plan".

3. Confirm that the **状态** is *准备好*.

4. Click **传输**.
   
    {{< figure src="/attachments/developerportal/deploy/tencent-deploy/image15.png" >}}

5. Change any constants in the **常量** tab: select the constant you want to edit and then click **编辑**.

6. Toggle any scheduled events in the **计划事件** tab: select the scheduled event you want to enable or disable and click **启停**.

7. Click **继续** to continue to the Start Application confirmation page.

8. Click **应用更改** to deploy the application to the selected environment. The app will start automatically once the deployment is successful.

You will be taken to the Environment Details page for the selected environment. You can find information about managing your environment from this page in [Managing Your Environments from the Environment Details Page](#environment-details), below.

## 5 Environments Page

The **环境概览** page of the ATM contains three sections:

* **部署包仓库**
* **环境**
* **活动日志**

### 5.1 Deployment Package Repository

The **部署包仓库** section lists the deployment packages which have been created for this app.

{{< figure src="/attachments/developerportal/deploy/tencent-deploy/image17.png" >}}

There are five buttons:

> * **刷新**

Reviewer: Sorry, I missed this one. 

The first button is actually **删除** (Delete), instead of **刷新** (Refresh), so 5.1.1 also needs to be changed accordingly.

Author: So we need to add a new section (5.1.1 Delete) and renumber all the other sections?
And add it here:
```suggestion
* **删除**
* **刷新**
```

Reviewer: On my screen, these is no **Refresh** button somehow – there is only the **Delete** button. Could you please check if your screen still shows the **Refresh** button? If not, then we should replace the current **5.1.1 Refresh** section with **5.1.1 Delete** section. If yes, maybe the button just changes according to the context.

1. Click **部署** next to the deployment package you wish to deploy.

2. Confirm the **目标** environment (you can select a different one here if one is available).

3. Confirm that the **状态** is *准备好*.

4. Click **传输**.
   
    {{< figure src="/attachments/developerportal/deploy/tencent-deploy/image15.png" >}}

5. Change any constants in the **常量** tab: select the constant you want to edit and then click **编辑**.

6. Toggle any scheduled events in the **计划事件** tab: select the scheduled event you want to enable or disable and click **启停**.

7. Click **继续** to continue to the Start Application confirmation page.

8. Click **应用更改** to deploy the application to the selected environment. The app will start automatically once the deployment is successful.

You will be taken to the Environment Details page for the selected environment. You can find information about managing your environment from this page in [Managing Your Environments from the Environment Details Page](#environment-details), below.

{{% todo %}}Continue from here{{% /todo %}}

## 5 Environments Page

The **环境概览** page of the ATM contains three sections:

> * Deployment Package Repository

Author: @Luyao-Zhang-1 
I'm a bit stuck on what to do here.
Should I translate these to be the same as the screen - then I also need to translate the section headings below which mirror these.
I feel that I should - but then I think we need to translate a lot more stuff!

Can you have a quick look at what I've done so far and see if you can come up with any guidance from the point of view of a reader who is bilingual?

(NB - it is not finished yet, so there may be some inconsistencies).

Thanks

Author: Decision:
1. Sections headings remain in English.
2. Where section headings relate to UX, introduce the Chinese term for the UX. "The **XXX** tab/button/page …"
3. Direct UX references in Chinese (and bold).
4. Use English when not referring directly to a UX element.

There are five buttons:

{{% todo %}}Should these be in English or in Chinese - and the sections (5.1.1 etc.) below?{{% /todo %}}

* 刷新
* 从 Team Server 创建包
* 上传
* 详情

> * Deploy

Reviewer: **Deploy** here would be **部署**.

Author: Thanks - missed that one. I'll **bold** them as well as they are UX buttons.

#### 6.1.2 Environment Details > Status

The **状态** status in **环境详情** shows you the status of the environment and is the same as the status shown on the Environments page and described in [Environment Status Indicators](#environment-status), above.

#### 6.1.3 Action Buttons

There are also buttons which allow you to perform various actions on your app and environment. These are described in the sections below.

##### 6.1.3.1 Stop/Start Application

If the app is not currently running (**运行时** is shown in gray) you will see **启动应用**. Clicking this will immediately trigger the app to begin running by increasing the number of replicas.

If the app is currently running, clicking **停止应用** immediately stops the application by reducing the number of replicas to zero.

##### 6.1.3.2 Transport Package

The **传输包** button allows you to deploy the deployment package in the current environment to another environment within the app, or to redeploy it in the current environment. See [Deploying the Deployment Package](#deploy-package), above, for more information.

##### 6.1.3.3 Scale Application

The **伸缩应用** button allows you to scale your app by increasing the number of replicas.

> To apply the new value, click **Restart the App and Scale**. Because you restart your app, it will be unavailable for a short time.

Reviewer: <img width="610" alt="image" src="https://user-images.githubusercontent.com/83757403/181509226-450c24ed-9ba3-454f-8f39-13b683c64f2f.png">
**Restart the App and Scale** in the Chinese version is **伸缩**.

Author: So just do:
```suggestion
To apply the new value, click **伸缩**. Because you restart your app, it will be unavailable for a short time.
```
?

For each environment, you can see a summary of the status of the resources and details of the package which is running in the environment (if there is one).

You can use one of the following buttons:

* **创建环境**
* **详情**
* **操作**

These are described in more detail, below.

In addition, there are several indicators describing the status of the environment. These are described in [Environment Status Indicators](#environment-status), below.

#### 5.2.1 Add Environment

> This adds a new environment as described in [Creating an Environment](#create-environment), above.

Reviewer: Although I use **我的应用** instead of **My Apps** as a section title, I am still wondering whether it is better to use Chinese or English for section titles.

Maybe we can discuss this next week?

Reviewer: If we choose to use English for section titles, then we probably need to mention the Chinese button name at the beginning of the section to show Chinese users which button we talk about in this section (and avoid immediately referring to the button from the start as "this/this button".)

Author: Yes - this is the thing which I most want to discuss at our meeting!

#### 5.2.2 Details

> This opens the **环境详情** page which is described in more detail in [Managing Your Environments from the Environment Details Page](#environment-details), below.

Reviewer: Same as above.

Author: Is this now resolved?

#### 5.2.3 Actions

> This button contains a list of actions which you can perform quickly on the environment. Most of these actions will be disabled if the app is currently starting or stopping. These actions are:

Reviewer: Same as above.

##### 5.2.4.1 Network

The **网络** indicator has the following values:

* Tick – the network is operational
* Cross – the network is not operational
* Spinner – the network is being provisioned

##### 5.2.4.2 Storage

The **存储** indicator has the following values:

* Tick – storage is provisioned
* Cross – storage is not provisioned
* Spinner – storage is being provisioned

##### 5.2.4.3 Database

The **数据库** indicator has the following values:

* Tick – the database is provisioned
* Cross – the database is not provisioned
* Spinner – the database is being provisioned

##### 5.2.4.4 Development

The word **开发环境** indicates that this environment is set up for development.

The word changes to **生产环境** if the environment is set up for production.

See [Creating an Environment](#create-environment), above, for more information.

### 5.3 Activity

This section shows all the activities, such as creating environments and deploying apps, which have taken place in this environment. You can sort the activities in either descending or ascending date and time order.

## 6 Managing Your Environments from the Environment Details Page{#environment-details}

> Each environment you create has an **Environment Details** page which allows you to monitor and manage your environments. You can reach this by clicking the **详情** button next to the environment you want to manage.

Reviewer: Is **Environment Details** written in English on the UI?

Author: No - it is called **环境详情** Thanks for spotting that I'd missed that.

If you have any outstanding changes to your environment the page will display a warning message. If you click **应用更改**, the environment will be stopped and restarted.

{{< figure src="/attachments/developerportal/deploy/tencent-deploy/image21.png" >}}

The environment details page consists of seven tabs:

> * 常规设置

Reviewer: Should the tab names be in bold?

* 模型选项
* 网络
* 运行时
* 日志级别
* TLS
* 调试器

These tabs are described below.

> 

Reviewer: Below, for the first sentence in 6.1, maybe change "The general tab shows information about your running app." to "The **常规设置** tab shows information..."?

Reviewer: Below, for the first sentence in 6.1.1 and 6.1.2, maybe change "This" to the Chinese UI text, because that's what Chinese users see on screen?

Author: I think so - but lets discuss it at our meeting?

> ##### 6.1.3.1 Stop/Start Application

Reviewer: For the first sentence in 6.1.1, 6.1.2, 6.1.3.2, 6.1.3.4, 6.1.3.4, 5.1.3.5, 6.1.3.6, 6.1.3.7, maybe change "This" to the Chinese UI text, because that's what Chinese users actually see on screen?

Author: I'll leave it for now - but probably change it after our meeting.

If the environment cannot be deleted, you will receive a warning, but can go ahead and remove it from the ATM.

##### 6.1.3.7 Change Purpose

This enables you to change the purpose of your app environment. You can label an environment as one used for development of the app, for example acceptance testing. In this case choose **开发环境** and the app will be deployed as an unlicensed App.

For production deployment, select **生产环境**. If you select **生产环境**, then you will be asked for the Subscription Secret which ensures that your app runs as a licensed app. For the differences between unlicensed/test apps and licensed apps, see the [Free App](/developerportal/deploy/mendix-cloud-deploy/#free-app) section in *Mendix Cloud*.

{{% alert color="warning" %}}
Your app can only be deployed to a production environment if security is set on. You will not receive an error if security is set off, but the deployment will appear to hang with a spinner being displayed.
{{% /alert %}}

##### 6.1.3.8 Change Subscription Secret{#change-subscription-secret}

> If you select Production as the **目的** of the app environment, then you will need to use a Subscription Secret which ensures that your app runs as a licensed app. If you need to enter or change the subscription secret, then you can do that here.

Reviewer: Should "Production" be italicized?

Author: Or should it be the Chinese? (生产环境).
Or perhaps better to say
> If you select production as the purpose of the app environment…

Let's discuss how much we need to translate and how much we will leave to the automatic translation?

Author: Keep as production

* Enable TLS – is TLS enabled for this environment?

* Use existing TLS secret or add new tls private key and certificate?

* Existing Secret Name – if you choose an existing TLS secret then you will be asked to enter its name

* TLS Private Key File and TLS Certificate File – if you choose to add a new key you will need to upload these two files

> 

Reviewer: The Chinese UI strings are as follows:

**启用 TLS** – is TLS enabled for this environment?
**使用现有的 TLS 密钥或添加新的 tls 私钥和证书？** – Use existing TLS secret or add new tls private key and certificate?
**现有的密钥名称** – if you choose an existing TLS secret then you will be asked to enter its name
**TLS 私钥文件** and **TLS 证书文件** – if you choose to add a new key you will need to upload these two files

Author: See proposals (above) for updating this section.

Click **保存并应用** to confirm your new configuration.

{{% todo %}}Is it still necessary to say:
You will receive a warning that you have made some changes. Click **应用更改** to restart the app and apply the changes.
{{% /todo %}}

### 6.7 Debugger

> On the **调试器** tab you can set up and view the credentials you need to debug your app when it is running on Tencent.

Reviewer: Is it needed to document the **启用调试器** (Enable Debugger) button here? If you click the button, you can set the type of the password to be either **自动生成** (generated automatically) or **自定义** (custom).

Author: Sounds like a good idea. Do you want to have a go, or shall I pick it up next week?

Reviewer: I added the info. It's a short piece.

## 7 Current Limitations{#limitations}

### 7.1 Reserved Names for Siemens Low Code Platform Apps{#reserved-names}

All names beginning **mendix-** are reserved for use by the Tencent cluster.

### 7.2 Deleting Your App

Delete all environments before you delete an app. If you delete an app which has environments on Tencent, you will not be able to reach the environments through the ATM.

### 7.3 App Security and Production

> If you attempt to deploy an app with security not set to production into a production environment you will not get an error, however the deployment will appear to hang with **个副本正在运行** and **运行时** showing a spinner.

Reviewer: Could we double-check this **个副本正在运行** here? It doesn't sound natural with the sentence here.

Author: Can you check with the Panda team on Slack? I think you will be able to get to the right answer quicker than I can in this case 😉

Reviewer: I will check this with the Panda team when I am back from PTO.

> * Enable TLS – is TLS enabled for this environment?

Author: ```suggestion
* **启用 TLS** – is TLS enabled for this environment?
```

> * Use existing TLS secret or add new tls private key and certificate?

Author: ```suggestion
* **使用现有的 TLS 密钥或添加新的 tls 私钥和证书** – Use existing TLS secret or add new TLS private key and certificate?
```

> * Existing Secret Name – if you choose an existing TLS secret then you will be asked to enter its name
> 
> * TLS Private Key File and TLS Certificate File – if you choose to add a new key you will need to upload these two files

Author: ```suggestion
* **现有的密钥名称** – if you choose an existing TLS secret then you will be asked to enter its name

* **TLS 私钥文件** and **TLS 证书文件** – if you choose to add a new key you will need to upload these two files
```
